06_gpu_and_ml/stable_diffusion/image_to_image.py

```diff
@@ -90,25 +90,57 @@ class Model:
     @modal.enter()
     def enter(self):
         print(f"Downloading {MODEL_NAME} if necessary...")
+
+        dtype = torch.bfloat16
+
+        self.seed = 42
+        self.device = "cuda"
+
+        transformer = FluxTransformer2DModel.from_single_file(
+            "https://huggingface.co/Comfy-Org/flux1-kontext-dev_ComfyUI/blob/main/split_files/diffusion_models/flux1-dev-kontext_fp8_scaled.safetensors",
+            torch_dtype=dtype,
+            cache_dir=CACHE_DIR,
+        )
+        quantize(transformer, weights=qfloat8)
+        freeze(transformer)
+
+        text_encoder_2 = T5EncoderModel.from_pretrained(
+            MODEL_NAME,
+            revision=MODEL_REVISION,
+            subfolder="text_encoder_2",
+            torch_dtype=dtype,
+            cache_dir=CACHE_DIR,
+        )
+        quantize(text_encoder_2, weights=qfloat8)
+        freeze(text_encoder_2)
+
         self.pipe = FluxKontextPipeline.from_pretrained(
             MODEL_NAME,
-            revision="f9fdd1a95e0dfd7653cb0966cda2486745122695",
-            torch_dtype=torch.bfloat16,
-            device_map="balanced",
+            revision=MODEL_REVISION,
+            transformer=transformer.to(self.device),
+            text_encoder_2=text_encoder_2.to(self.device),
+            torch_dtype=dtype,
             cache_dir=CACHE_DIR,
+            device_map="balanced",
```

Comment:
I am struggling to understand if this is good or not. My understanding is that this will split the model evenly between GPUs. But in this case we only have one GPU. Why do we need to set this?